modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java

```diff
@@ -1077,8 +1059,17 @@ public UserAccount putUserAccountByExternalReferenceCode(
 			_getAddresses(null, userAccount),
 			_getServiceBuilderEmailAddresses(null, userAccount),
 			_getServiceBuilderPhones(null, userAccount),
-			_getWebsites(null, userAccount), false,
-			_createServiceContext(userAccount));
+			_getWebsites(null, userAccount), false, serviceContext);
+
+		Integer workflowStatus = _getWorkflowStatus(
+			userAccount.getStatusAsString());
+
+		if (workflowStatus == null) {
+			throw new BadRequestException("Status is invalid");
+		}
+
+		user = _userService.updateStatus(
+			user.getUserId(), workflowStatus, serviceContext);
```

Comment:
probably also in this case is better to validate the status before doing all the stuff. 
Validate it before and update it after. What do you think?